src/logics/Theory.h

```diff
@@ -173,14 +173,16 @@ class Theory
     virtual bool            simplify(const vec<PFRef>&, PartitionManager& pmanager, int) = 0; // Simplify a vector of PushFrames in an incrementality-aware manner
     SubstitutionResult      computeSubstitutions(PTRef fla);
     void                    printFramesAsQuery(const vec<PFRef> & frames, std::ostream & s) const;
-    virtual                ~Theory()                           {};
+    virtual bool            okToPartition(PTRef)    = 0;
```

Comment:
```suggestion
    virtual bool            okToPartition(PTRef)  { return true; }
```
How about having the default implementation? Then we do not need to add it to the `UFTheory` and `CUFTheory`.

Reply:
That's a good idea.